ApplicationLibCode/ReservoirDataModel/RigMainGrid.cpp

```diff
@@ -264,13 +264,28 @@ void RigMainGrid::setDisplayModelOffset( cvf::Vec3d offset )
 /// Compute cell ranges for active and valid cells
 /// Compute bounding box in world coordinates based on node coordinates
 //--------------------------------------------------------------------------------------------------
-void RigMainGrid::computeCachedData()
+void RigMainGrid::computeCachedData( std::string* aabbTreeInfo )
 {
     initAllSubGridsParentGridPointer();
     initAllSubCellsMainGridCellIndex();
 
     m_cellSearchTree = nullptr;
-    buildCellSearchTree();
+
+    const double maxNumberOfLeafNodes = 4000000;
+
+    double factor = cellCount() / maxNumberOfLeafNodes;
+    factor        = std::ceil( factor );
+
+    size_t cellsPerBoundingBox = static_cast<size_t>( factor );
+    cellsPerBoundingBox        = std::max( size_t( 1 ), cellsPerBoundingBox );
```

Comment:
Both factor and cellsPerBoundingBox can be const if the computation is done as one expression.
`const double factor = std::ceil(cellCount()/maxNumberOfLeafNodes);`